### PR TITLE
Add *.pdf preview

### PR DESCRIPTION
**Pull request by DiFly**
> **Note**
> Add *.pdf preview in modal iframe

# Description (需求描述)
Added the functionality of previewing *.pdf files by clicking on the file icon in the table or list. 

# Motivation (需求背景)
We need a convenient preview of *.pdf files without downloading to a local PC.

# Key changes (关键改动）
This is implemented by creating a simple modal window with an iframe (the source of which is a link to the *.pdf file). Additional libraries are not used.

## Underlying risk (潜在风险)
If the browser does not support native viewing of *.pdf files, the file will not be viewed.

# Showcase (结果展示）
Screenshots of the new feature.
Screenshot_1
Screenshot_2

### CODE REVIEW

File: packages/core/client/src/schema-component/antd/upload/shared.ts [hunk 1]
```diff
@@ -14,6 +14,11 @@ export const isImage = (extName: string) => {
   return reg.test(extName);
 };
 
+export const isPdf = (extName: string) => {
+  const reg = /\.(pdf|PDF)$/;
+  return reg.test(extName);
```
Comment (dream2023) [inline packages/core/client/src/schema-component/antd/upload/shared.ts:19]: `extName.toLowerCase().endsWith('.pdf')` is better.

File: packages/core/client/src/schema-component/antd/upload/ReadPretty.tsx [hunk 2]
```diff
@@ -25,7 +25,13 @@ ReadPretty.File = function File(props: UploadProps) {
   const images = toImages(toArr(value));
   const [photoIndex, setPhotoIndex] = useState(0);
   const [visible, setVisible] = useState(false);
+  const [visiblePDF, setVisiblePDF] = useState(false);
```
Comment (dream2023) [inline packages/core/client/src/schema-component/antd/upload/ReadPretty.tsx:28]:
```ts
const [fileType, setFileType] = useState<'image' | 'pdf'>();
```

File: packages/core/client/src/schema-component/antd/upload/ReadPretty.tsx [hunk 3]
```diff
@@ -25,7 +25,13 @@ ReadPretty.File = function File(props: UploadProps) {
   const images = toImages(toArr(value));
   const [photoIndex, setPhotoIndex] = useState(0);
```
Comment (dream2023) [inline packages/core/client/src/schema-component/antd/upload/ReadPretty.tsx:26]:
```ts
const [fileIndex, setFileIndex] = useState(0);
```

File: packages/core/client/src/schema-component/antd/upload/ReadPretty.tsx [hunk 4]
```diff
@@ -38,6 +44,11 @@ ReadPretty.File = function File(props: UploadProps) {
                 e.stopPropagation();
                 setVisible(true);
                 setPhotoIndex(index);
```
Comment (dream2023) [inline packages/core/client/src/schema-component/antd/upload/ReadPretty.tsx:46]:
```ts
setVisible(true);
setFileIndex(index);
setFileType('image')
```

File: packages/core/client/src/schema-component/antd/upload/ReadPretty.tsx [hunk 5]
```diff
@@ -38,6 +44,11 @@ ReadPretty.File = function File(props: UploadProps) {
...
+              } else if(isPdf(file.extname)) {
+                e.preventDefault();
+                e.stopPropagation();
+                setVisiblePDF(true);
+                setPhotoIndex(index);
```
Comment (dream2023) [inline packages/core/client/src/schema-component/antd/upload/ReadPretty.tsx:51]:
```ts
setVisible(true);
setFileIndex(index);
setFileType('pdf')
```

File: packages/core/client/src/schema-component/antd/upload/ReadPretty.tsx [hunk 6]
```diff
@@ -132,6 +143,61 @@ ReadPretty.File = function File(props: UploadProps) {
           ]}
         />
       )}
+      
+      {visiblePDF && (
```
Comment (dream2023) [inline packages/core/client/src/schema-component/antd/upload/ReadPretty.tsx:147]:
```tsx
visible && fileType === 'pdf' && 
```

File: packages/core/client/src/schema-component/antd/upload/ReadPretty.tsx [hunk 7]
```diff
@@ -132,6 +143,61 @@ ReadPretty.File = function File(props: UploadProps) {
...
+            maxWidth: '90vw',
+            maxHeight: 'calc(100vh - 112px)',
+            height: '90vh',
+            width: '90vw',
+            background: 'white',
+            display: 'flex',
+            flexDirection: 'column',
+            alignItems: 'center',
+            overflowY: 'auto'
+          }} >
```
Comment (dream2023) [inline packages/core/client/src/schema-component/antd/upload/ReadPretty.tsx:169]: Replace by [modal](https://4x.ant.design/components/modal/) Component

**Review by dream2023 — CHANGES_REQUESTED** [review]

File: packages/core/client/src/schema-component/antd/upload/Upload.tsx [hunk 8]
```diff
@@ -189,6 +196,60 @@ Upload.Attachment = connect((props: UploadProps) => {
...
+                  marginRight: '4px'
+                }}
+                onClick={(e) => {
+                  e.preventDefault();
+                  e.stopPropagation();
+                  const file = images[photoIndex];
+                  saveAs(file.url, `${file.title}${file.extname}`);
+                }}
+              >
+                Download
```
Comment (dream2023) [inline packages/core/client/src/schema-component/antd/upload/Upload.tsx:243]:
```ts
const { t } = useTranslation();
t('download')
t('close')
```

**Review by dream2023 — CHANGES_REQUESTED** [review]

**Comment by DiFly** [comment]
Refactoring done, are there any more changes to make?

**Pull request merged**